google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -137,11 +137,16 @@ public CancellableRunnable createDeployRunner(
         targetEnvironment,
         wrapCallbackForUsageTracking(callback, deploymentConfiguration, targetEnvironment));
 
-    if (targetEnvironment.isStandard()
-        || (targetEnvironment.isFlexible() && AppEngineUtil.isFlexCompat(project, source))) {
-      return createStandardRunner(loggingHandler, source.getFile(), deploy);
+    if (targetEnvironment.isStandard()) {
```

Comment:
I'd need to see it to confirm, but I think all this code will be easier to read if you restore it to the way it was, then extract out a local `isFlexCompat` variable to use as needed. This gets rid of the nested conditionals and the comments you added to the unnamed parameters.